extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/runtime/KeycloakPolicyEnforcerUtil.java

```diff
@@ -51,18 +52,20 @@ static PolicyEnforcer createPolicyEnforcer(OidcTenantConfig oidcConfig,
         adapterConfig.setResource(oidcConfig.getClientId().get());
         adapterConfig.setCredentials(getCredentials(oidcConfig));
 
-        boolean trustAll = oidcConfig.tls.getVerification().isPresent()
-                ? oidcConfig.tls.getVerification().get() == OidcCommonConfig.Tls.Verification.NONE
-                : tlsConfigTrustAll;
-        if (trustAll) {
-            adapterConfig.setDisableTrustManager(true);
-            adapterConfig.setAllowAnyHostname(true);
-        } else if (oidcConfig.tls.trustStoreFile.isPresent()) {
-            adapterConfig.setTruststore(oidcConfig.tls.trustStoreFile.get().toString());
-            adapterConfig.setTruststorePassword(oidcConfig.tls.trustStorePassword.orElse("password"));
-            if (OidcCommonConfig.Tls.Verification.CERTIFICATE_VALIDATION == oidcConfig.tls.verification
-                    .orElse(OidcCommonConfig.Tls.Verification.REQUIRED)) {
+        if (!tlsConfigSupport.useTlsRegistry()) {
+            boolean trustAll = oidcConfig.tls.getVerification().isPresent()
+                    ? oidcConfig.tls.getVerification().get() == OidcCommonConfig.Tls.Verification.NONE
+                    : tlsConfigSupport.isGlobalTrustAll();
+            if (trustAll) {
+                adapterConfig.setDisableTrustManager(true);
                 adapterConfig.setAllowAnyHostname(true);
```

Comment:
Is it the hostname verification? I guess so, but there are so many names referring to the same thing.

Reply:
@cescoffier I don't think you need to worry about this code, all I did is to put it behind flag `!useTlsRegistry` and I take global trust all flag from `tlsConfigSupport.isGlobalTrustAll()` instead of registry, but it's the same source of information. So this is the old behavior.

What matters is the code you commented below in the KeycloakPolicyEnforcerUtil.

Reply:
but if you are interested, this original behavior did create SSL context that allowed all host names and `javax.net.ssl.X509TrustManager` that always trusted (NOOP)